app/src/main/java/ru/chaichuk/hwapp/api_v3/MovieDbApi.kt

```diff
@@ -58,6 +58,38 @@ class MovieDbApi {
             }
     }
 
+    suspend fun getMovieByIdFromNet(id : Int) : Movie? {
+        var movie : Movie? = null
+        val movieDTO = getMovieDetailsWithCredits(id)
+
+        if (movieDTO != null) {
+            movie =         Movie(
+                movieDTO.id.toInt(),
+                movieDTO.title,
+                movieDTO.overview,
+                BuildConfig.BASE_IMAGE_W780_URL + movieDTO.posterPath,
+                BuildConfig.BASE_IMAGE_W780_URL + movieDTO.backdropPath,
+                movieDTO.voteAverage.toFloat(),
+                movieDTO.voteCount.toInt(),
+                if (movieDTO.adult) 16 else 13,
+                movieDTO.runtime.toInt(),
+                movieDTO.genres.map { Genre(it.id.toInt(), it.name) },
+                movieDTO.credits.cast.mapNotNull { castDTO ->
+                    castDTO.profilePath?.let { picture ->
+                        Actor(
+                            castDTO.id.toInt(),
+                            castDTO.name,
+                            BuildConfig.BASE_IMAGE_W200_URL + picture
+                        )
+                    }
+                },
+                true
+            )
+        }
+
+        return movie
+    }
+
     private suspend fun getPopularMovies(page : Int) : List <MovieDTO> = withContext(Dispatchers.IO) {
         val moviesPage = RetrofitModule.moviesApi.popularMovies(page = page).log("getPopularMovies api item")
         return@withContext moviesPage.movies
```

Comment:
I think you can use just `moviesPage.movies` withOut return

Reply:
changed function body to
RetrofitModule.moviesApi.popularMovies(page = page).log("getPopularMovies api item").movies
without return